executor/src/test/java/com/redhat/service/bridge/executor/ExecutorTest.java

```diff
@@ -0,0 +1,120 @@
+package com.redhat.service.bridge.executor;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.redhat.service.bridge.actions.ActionInvoker;
+import com.redhat.service.bridge.actions.ActionProvider;
+import com.redhat.service.bridge.actions.ActionProviderFactory;
+import com.redhat.service.bridge.actions.kafkatopic.KafkaTopicAction;
+import com.redhat.service.bridge.infra.models.actions.BaseAction;
+import com.redhat.service.bridge.infra.models.dto.BridgeDTO;
+import com.redhat.service.bridge.infra.models.dto.BridgeStatus;
+import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
+import com.redhat.service.bridge.infra.models.filters.BaseFilter;
+import com.redhat.service.bridge.infra.models.filters.StringEquals;
+import com.redhat.service.bridge.infra.utils.CloudEventUtils;
+
+import io.cloudevents.CloudEvent;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ExecutorTest {
```

Comment:
Currently the Executor is not covered by any test/integration tests (because we are still missing the integration test to assess that the KafkaTopicAction worked properly). 

This test is checking that the `onEvent` is calling the action invoker after the filtering and the templating operations. It is mocking "only" the actions factories to intercept the call. 

From my perspective, it is okaish. But no strong opinions from my side on this. If you prefer, I can remove it @danielezonca

Reply:
Fine for me I think it is just important to keep the mocking under control and limit it as much as we can 👍